conn/pool.go

```diff
@@ -344,10 +344,11 @@ func (p *Pool) IsHealthy() bool {
 
 // HealthInfo returns the healthinfo.
 func (p *Pool) HealthInfo() pb.HealthInfo {
-	p.RLock()
-	defer p.RUnlock()
+	ok := p.IsHealthy()
```

Comment:
reading before acquiring lock?

Reply:
p.IsHealthy() takes the read lock internally.